tests/main.ml

```diff
@@ -78,6 +78,32 @@ let test_readv () =
   check_string ~__POS__ ~expected:"est fil" (Bigstringaf.to_string b2);
   ()
 
+let test_finalise () =
+  let () =
+    (* Finalising an unclosed Uring is successful. *)
+    let finalised = ref false in
+    let () =
+      Uring.create ~default:() ~queue_depth:1 ()
+      |> Gc.finalise_last (fun _ -> finalised := true)
+    in
+    Gc.full_major ();
+    assert_ ~__POS__ !finalised;
+    Gc.full_major ()
```

Comment:
I have no idea why, but the C finaliser appears to only run on this _second_ forced GC major, even though the first one triggers the OCaml finaliser...